src/Simulation/Native/src/external/fused.hpp

```diff
@@ -41,7 +41,14 @@ class Fused
       fusedgates = Fusion();
     }
 
+    Fusion getFusedGates() const {
```

Comment:
You bring out a very interesting point here, and I noticed that you brought this up in multiple places. However, making it a reference takes us away from the getFoo/setFoo design pattern used throughtout the repository. This change would require a significant overhaul in many aspects of the codebase. We could look into making use of std:move, however it would be appropriate to get back to this as part of a another PR linking it to an appropriate issue. Given that the data structures these work on are set of gates, I believe copying does not add much overhead. right now. Thank you for pointing this out!